src/SFA.DAS.AssessorService.Application.Api.UnitTests/Controllers/Register/Query/SearchOrganisationBySearchStringTests.cs

```diff
@@ -0,0 +1,82 @@
+using SFA.DAS.AssessorService.Api.Types.Models;
+using Microsoft.AspNetCore.Mvc;
+using FluentAssertions;
+
+namespace SFA.DAS.AssessorService.Application.Api.UnitTests.Controllers.Register.Query
+{
+    [TestFixture]
+    public class SearchOrganisationBySearchStringTests
+    {
+        private static RegisterQueryController _queryController;
+        private static object _result;
+        private static Mock<IMediator> _mediator;
+        private static Mock<ILogger<RegisterQueryController>> _logger;
+        private List<AssessmentOrganisationSummary> _expectedAssessmentOrganisationSetOfDetails;
+        private AssessmentOrganisationSummary _assOrgDetails1;
+        private AssessmentOrganisationSummary _assOrgDetails2;
+        private string searchString = "Test";
+
+
+        [SetUp]
+        public void Arrange()
+        {
+            _mediator = new Mock<IMediator>();
+            _logger = new Mock<ILogger<RegisterQueryController>>();
+            // needs more details
+            _assOrgDetails1 = new AssessmentOrganisationSummary { Id = "Id1", Name = "Test 9", Ukprn = 9999999 };
+            _assOrgDetails2 = new AssessmentOrganisationSummary  {Id = "Id2", Name = "Test 2", Ukprn = 8888888 };
+            _expectedAssessmentOrganisationSetOfDetails = new List<AssessmentOrganisationSummary>
+            {
+                _assOrgDetails1,
+                _assOrgDetails2
+            };
+
+            _mediator.Setup(m =>
+                m.Send(It.IsAny<SearchAssessmentOrganisationsRequest>(),
+                    new CancellationToken())).ReturnsAsync(_expectedAssessmentOrganisationSetOfDetails);
+            _queryController = new RegisterQueryController(_mediator.Object, _logger.Object);
+            _result = _queryController.SearchAssessmentOrganisations(searchString).Result;
+        }
+
+
+        [Test]
+        public void SearchAssessmentOrganisationsByStandardIdReturnExpectedActionResult()
+        {
+            _result.Should().BeAssignableTo<IActionResult>();
+        }
+        [Test]
+        public void MediatorSendsExpectedSearchAssessmentOrganisationsBySsarchstringRequest()
+        {
+            _mediator.Verify(m => m.Send(It.IsAny<SearchAssessmentOrganisationsRequest>(), new CancellationToken()));
+        }
+        [Test]
+        public void SearchAssessmentOrganisationsReturnOk()
+        {
+            _result.Should().BeOfType<OkObjectResult>();
+        }
+        [Test]
+        public void ResultsAreOfTypeListAssessmentOrganisationDetails()
+        {
+            ((OkObjectResult)_result).Value.Should().BeOfType<List<AssessmentOrganisationSummary>>();
+        }
+        [Test]
+        public void ResultsMatchExpectedListOfAssessmentOrganisationDetails()
+        {
+            var organisations = ((OkObjectResult)_result).Value as List<AssessmentOrganisationSummary>;
+            organisations.Count.Should().Be(2);
+            organisations.Should().Contain(_assOrgDetails1);
```

Comment:
Picky point but I am wondering whether or not we should have a single test per assert?

Reply:
🤔 Personally I think in this case it's valid as the two asserts are v closely related.  But generally, I'd agree. 👍

Reply:
Yep 👍

Reply:
Hmm, this is tricky, because I'm really only concerned that the list is full and contains all elements.  As ordering is not part of the criteria, it should pass whatever the order of the items.  If ordering WERE part of the criteria, I'd do an assert comparing the lists, but as it's not I can't do that.... so I see the 3 asserts here are actually checking one thing, and by breaking them apart into separate tests, I'm fracturing what I'm trying to test.... does that make sense?  If any assert fails, it fails my test rather than my 'tests', as it were.